cpp/simulations/abm_braunschweig.cpp

```diff
@@ -0,0 +1,711 @@
+    uint32_t count                        = 0;
+    std::map<std::string, uint32_t> index = {};
+    for (auto const& title : titles) {
+        index.insert({title, count});
+        row_string.push_back(title);
+        count++;
+    }
+    std::map<uint32_t, mio::abm::LocationId> locations = {};
+    std::map<uint32_t, mio::abm::Person&> persons      = {};
+
+    int number_persons = 0;
+
+    auto hospital = world.add_location(mio::abm::LocationType::Hospital);
+    world.get_individualized_location(hospital).get_infection_parameters().set<mio::abm::MaximumContacts>(5);
+    world.get_individualized_location(hospital).set_capacity(584, 26242);
+    auto icu = world.add_location(mio::abm::LocationType::ICU);
+    world.get_individualized_location(icu).get_infection_parameters().set<mio::abm::MaximumContacts>(5);
+    world.get_individualized_location(icu).set_capacity(30, 1350);
+
+    while (std::getline(fin, line) && number_persons < 10000) {
+        row.clear();
+
+        // read columns in this row
+        split_line(line, &row);
+        line.erase(std::remove(line.begin(), line.end(), '\r'), line.end());
+
+        uint32_t person_id          = row[index["puid"]];
+        uint32_t age                = row[index["age"]]; // TODO
+        uint32_t target_location_id = std::abs(row[index["loc_id_end"]]);
+        uint32_t start_location_id  = std::abs(row[index["loc_id_start"]]);
+        uint32_t home_id            = row[index["huid"]];
+        uint32_t activity_end       = row[index["activity_end"]];
+        uint32_t trip_start         = row[index["start_time"]];
+
+        mio::abm::LocationId home;
+        auto it_home = locations.find(home_id);
+        if (it_home == locations.end()) {
+            home = world.add_location(mio::abm::LocationType::Home, 1);
+            locations.insert({home_id, home});
+        }
+        else {
+            home = it_home->second;
+        }
+        auto it_person = persons.find(person_id);
+        if (it_person == persons.end()) {
+            auto& person = world.add_person(home, determine_age_group(age));
+            person.set_assigned_location(home);
+            person.set_assigned_location(hospital);
+            person.set_assigned_location(icu);
+            persons.insert({person_id, person});
+            it_person = persons.find(person_id);
+            number_persons++;
+        }
+        auto it_location = locations.find(target_location_id);
+        mio::abm::LocationId location;
+        if (get_location_type(activity_end) != mio::abm::LocationType::Home) {
+            if (it_location == locations.end()) {
+                location = world.add_location(get_location_type(activity_end), 1);
+                locations.insert({target_location_id, location});
+            }
+            else {
+                location = it_location->second;
+            }
+            it_person->second.set_assigned_location(location);
+        }
+        world.get_trip_list().add_trip(
```

Comment:
On windows this fails, due to start_location not being set. For a quick fix i set start_location in trips to the destination and not give that argument over. start_location  isn't used it in code either way